mlrun/runtimes/function.py

```diff
@@ -527,3 +519,106 @@ def get_auth_filled_platform_dashboard_url(dashboard: str) -> str:
+    spec.cmd = function.spec.build.commands or []
+    project = function.metadata.project or "default"
+    tag = function.metadata.tag
+    handler = function.spec.function_handler
+
+    # In Nuclio 1.6.0 default serviceType changed to "ClusterIP", make sure we're using NodePort
+    spec.set_config("spec.serviceType", "NodePort")
+    if function.spec.readiness_timeout:
+        spec.set_config("spec.readinessTimeoutSeconds", function.spec.readiness_timeout)
+    if function.spec.resources:
+        spec.set_config("spec.resources", function.spec.resources)
+    if function.spec.no_cache:
+        spec.set_config("spec.build.noCache", True)
+    if function.spec.replicas:
+        spec.set_config("spec.minReplicas", function.spec.replicas)
+        spec.set_config("spec.maxReplicas", function.spec.replicas)
+    else:
+        spec.set_config("spec.minReplicas", function.spec.min_replicas)
+        spec.set_config("spec.maxReplicas", function.spec.max_replicas)
+
+    dashboard = get_auth_filled_platform_dashboard_url(dashboard)
+    if function.spec.base_spec:
+        config = nuclio.config.extend_config(
+            function.spec.base_spec, spec, tag, function.spec.build.code_origin
+        )
+        update_in(config, "metadata.name", function.metadata.name)
+        update_in(config, "spec.volumes", function.spec.to_nuclio_vol())
+        base_image = get_in(config, "spec.build.baseImage")
+        if base_image:
+            update_in(config, "spec.build.baseImage", enrich_image_url(base_image))
+
+        logger.info("deploy started")
+        name = get_fullname(function.metadata.name, project, tag)
+        function.status.nuclio_name = name
+        update_in(config, "metadata.name", name)
+        return nuclio.deploy.deploy_config(
+            config,
+            dashboard,
+            name=name,
+            project=project,
+            tag=tag,
+            verbose=function.verbose,
+            create_new=True,
+            watch=watch,
+        )
+    else:
+
+        name, config, code = nuclio.build_file(
+            function.spec.source,
+            name=function.metadata.name,
+            project=project,
+            handler=handler,
+            tag=tag,
+            spec=spec,
+            kind=function.spec.function_kind,
+            verbose=function.verbose,
+        )
+
+        update_in(config, "spec.volumes", function.spec.to_nuclio_vol())
+        name = get_fullname(name, project, tag)
+        function.status.nuclio_name = name
+
+        update_in(config, "metadata.name", name)
+        return deploy_config(
+            config,
+            dashboard_url=dashboard,
+            name=name,
+            project=project,
+            tag=tag,
+            verbose=function.verbose,
+            create_new=True,
+            watch=watch,
+        )
+
+
+def get_nuclio_deploy_status(
+    name, project, tag, dashboard="", last_time=None, verbose=False
+):
+    api_address = find_dashboard_url(dashboard)
```

Comment:
Same, no need, use config

Reply:
this is something else, from nuclio-jupyter, from host to full path

Reply:
Should be changed to 
```suggestion
    api_address = find_dashboard_url(dashboard or mlconf.nuclio_dashboard_url)
```

Reply:
done